staging/src/k8s.io/kubectl/pkg/cmd/create/create_ingress.go

```diff
@@ -0,0 +1,82 @@
+package create
+
+import (
+	"github.com/spf13/cobra"
+	cmdutil "k8s.io/kubectl/pkg/cmd/util"
+	"k8s.io/cli-runtime/pkg/genericclioptions"
+	"k8s.io/kubectl/pkg/util/i18n"
+	"k8s.io/kubectl/pkg/util/templates"
+	generateversioned "k8s.io/kubectl/pkg/generate/versioned"
+	"k8s.io/kubectl/pkg/generate"
+)
+
+// IngressOpts is returned by NewCmdCreateIngress
+type IngressOpts struct {
+	CreateSubcommandOptions *CreateSubcommandOptions
+}
+
+// NewCmdCreateIngress is a macro command to create a new ingress.
+func NewCmdCreateIngress(f cmdutil.Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	options := &IngressOpts{
+		CreateSubcommandOptions: NewCreateSubcommandOptions(ioStreams),
+	}
+
+	cmd := &cobra.Command{
+		Use:                   "ingress NAME --host=hostname| --service-name=servicename [--service-port=serviceport] [--path=path] [--dry-run]",
+		DisableFlagsInUseLine: true,
+		Aliases:               []string{"ing"},
+		Short:                 i18n.T("Create an ingress with the specified name."),
+		Long:                  templates.LongDesc(i18n.T("Create an ingress with the specified name.")),
```

Comment:
Long and example are usually defined outside for readability.

Reply:
These are now created in two global cars